vibi-dpu/src/core/github/setup.rs

```diff
@@ -113,3 +114,54 @@ async fn process_prs(repo_owner_async: &String, repo_name_async: &String, access
     }
     
 }
+
+pub async fn setup_self_host_user_repos_github(access_token: &str) {
+    let repo_provider = env::var("PROVIDER").expect("provider must be set");
+
+    let repos_opt = get_user_accessed_github_repos(&access_token).await;
+    if repos_opt.is_none() {
+        log::error!("[setup_self_host_user_repos_github] No repositories found for the user");
+        return;
+    }
+    let repos = repos_opt.expect("Empty repos option");
+    log::debug!("[setup_self_host_user_repos_github] Got repos: {:?}", repos);
+
+    // Create a mapping between repo_owner and associated repo_names
+    let mut repo_owner_map: std::collections::HashMap<String, Vec<String>> = std::collections::HashMap::new();
+
+    for repo in repos {
+        let token_copy = access_token.clone();
+        let mut repo_copy = repo.clone();
+        clone_git_repo(&mut repo_copy, &token_copy, &repo_provider).await;
+        let repo_name = repo.name();
+        let repo_owner = repo.owner();
+        repo_owner_map
+            .entry(repo_owner.to_string())
+            .or_insert_with(Vec::new)
+            .push(repo_name.to_string());
```

Comment:
So this will insert a key with empty vector as value in the map, right?

Reply:
yess, firstly this will check if an entry with the same repo_owner exist else it will create an entry with corresponding vector values.